File: src/main/java/org/jcodec/containers/mp4/boxes/GenericMediaInfoBox.java
package org.jcodec.containers.mp4.boxes;

import js.nio.ByteBuffer;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * @author The JCodec project
 * 
 */
public class GenericMediaInfoBox extends FullBox {
    private short graphicsMode;
    private short rOpColor;
    private short gOpColor;
    private short bOpColor;
    private short balance;

    public static String fourcc() {
        return "gmin";
    }

    public static GenericMediaInfoBox createGenericMediaInfoBox() {
        return new GenericMediaInfoBox(new Header(fourcc()));
    }

    public GenericMediaInfoBox(Header atom) {
        super(atom);
    }

    public void parse(ByteBuffer input) {
        super.parse(input);
        graphicsMode = input.getShort();
        rOpColor = input.getShort();
        gOpColor = input.getShort();
        bOpColor = input.getShort();
        balance = input.getShort();
        input.getShort();
    }

    protected void doWrite(ByteBuffer out) {
        super.doWrite(out);
        out.putShort(graphicsMode);
        out.putShort(rOpColor);
        out.putShort(gOpColor);
        out.putShort(bOpColor);
        out.putShort(balance);
        out.putShort((short) 0);
    }
}

File: src/main/java/org/jcodec/containers/mp4/boxes/GenericMediaInfoBox.kt
package org.jcodec.containers.mp4.boxes

import java.nio.ByteBuffer

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * @author The JCodec project
 */
class GenericMediaInfoBox(atom: Header) : FullBox(atom) {
    private var graphicsMode: Short = 0
    private var rOpColor: Short = 0
    private var gOpColor: Short = 0
    private var bOpColor: Short = 0
    private var balance: Short = 0
    override fun parse(input: ByteBuffer) {
        super.parse(input)
        graphicsMode = input.short
        rOpColor = input.short
        gOpColor = input.short
        bOpColor = input.short
        balance = input.short
        input.short
    }

    override fun doWrite(out: ByteBuffer) {
        super.doWrite(out)
        out.putShort(graphicsMode)
        out.putShort(rOpColor)
        out.putShort(gOpColor)
        out.putShort(bOpColor)
        out.putShort(balance)
        out.putShort(0.toShort())
    }

    override fun estimateSize(): Int {
        return 24
    }

    companion object {
        @JvmStatic
        fun fourcc(): String {
            return "gmin"
        }

        @JvmStatic
        fun createGenericMediaInfoBox(): GenericMediaInfoBox {
            return GenericMediaInfoBox(Header(fourcc()))
        }
    }
}